crates/metaslang/bindings/src/lib.rs

```diff
@@ -79,11 +83,20 @@ impl<KT: KindTypes + 'static> Bindings<KT> {
             cursor_to_definitions: HashMap::new(),
             cursor_to_references: HashMap::new(),
             context: None,
+            built_ins_file: None,
         }
     }
 
+    pub fn add_built_ins(&mut self, tree_cursor: Cursor<KT>) {
+        assert!(self.built_ins_file.is_none(), "Built-ins already added");
+        let file = self.stack_graph.get_or_create_file(BUILT_INS_FILE_PATH);
+        _ = self.add_file_internal(file, tree_cursor);
+        self.built_ins_file = Some(file);
+    }
+
     pub fn add_file(&mut self, file_path: &str, tree_cursor: Cursor<KT>) {
```

Comment:
I was thinking the opposite thing: `add_file` should check that the user is not trying to add a file as the built-ins, or resolve that internally in some other way so that there's no possibility of clashing. But as I said in the previous comment, I guess it depends on who should handle the built-ins concept.